src/actions/vis-state-actions.js

```diff
@@ -508,6 +508,76 @@ export function setFeatures(features) {
   }
 }
 
+/**
+ * It will apply the provide feature as filter to the given layer.
+ * If the given feature is already applied as filter to the layer it  will remove it from  the filter list
+ * @memberof visStateActions
+ * @param {Object} featureId feature to use as filter base for layer
+ * @param {Object} layer
+ * @return {{feature: *, type: ActionTypes.TOGGLE_POLYGON_FILTER, layer: *}}
+ */
+export function togglePolygonFilter(layer, featureId) {
+  return {
+    type: ActionTypes.TOGGLE_POLYGON_FILTER,
+    layer,
+    featureId
+  };
+}
+
+/**
+ * Set the current feature to be edited/deleted
+ * @memberof visStateActions
+ * @param {Object} feature
+ * @param {string} feature.selectedFeatureId
+ * @return {{selectedFeatureId: *, type: ActionTypes.SET_SELECTED_FEATURE}}
+ */
+export function setSelectedFeature({selectedFeatureId}) {
```

Comment:
Seems like all the places where `onSelect` are called in `editor.js`, you are only passing `{selectedFeatureId: null | id}`,  why not unwrap the argument to be `selectedFeatureId`?

Reply:
Initially I thought I was going to propagate the full feature